cli/cli.js

```diff
@@ -194,3 +203,65 @@ function createJson(plans) {
         2,
     );
 }
+
+/**
+ * Runs the CLI Encode command
+ * @param {object} localOptions
+ * @param {Command} command
+ * @returns {Promise<void>}
+ */
+async function encode(localOptions, command) {
+    const globalParentOptions = command.parent.opts();
+    const { rpcUrl, fromBlock, pauseAddress } = globalParentOptions;
+
+    if (rpcUrl === defaults.RPC_URL) {
+        console.warn(
+            chalk.yellow(
+                `🟡 WARNING: Falling back to a public provider: ${rpcUrl}. For a better experience, set a custom RPC URL with the --rpc-url <rpc-url> option or the ETH_RPC_URL env variable.`,
+            ),
+        );
+    }
+
+    const spinner = ttyOnlySpinner().start("Fetching pending pause plans...");
+
+    try {
+        const pause = new ethers.Contract(
+            pauseAddress,
+            defaults.MCD_PAUSE_ABI,
+            ethers.getDefaultProvider(rpcUrl),
+        );
+
+        const status = "PENDING";
+        const plans = await fetchPausePlans(pause, { fromBlock, status });
+        spinner.success("Done!");
+
+        if (plans.length === 0) {
+            console.log(chalk.yellow("No pending spells found to encode."));
+            return;
```

Comment:
Maybe it's not that important, because it's a CLI used by humans, but it's a good practice to exit abnormally when the output is not in the same format as the expected one. In this case it's a JSON output, so someone could try to pipe it to a JSON parsing command for whatever reason.

```suggestion
            process.exit(1);
```